### ISSUE TITLE
Hit an issue when created my tables and publication and added select for all tables in sync_rules

### ISSUE THREAD

**Issue by MichalNemec** [powersync-ja/self-host-demo#6]
```
2024-04-24 03:15:02 {"level":"info","timestamp":"2024-04-24T01:15:02.950Z","message":"Booting"}
2024-04-24 03:15:02 {"level":"info","timestamp":"2024-04-24T01:15:02.969Z","message":"Starting system"}
2024-04-24 03:15:02 {"level":"info","timestamp":"2024-04-24T01:15:02.969Z","message":"System started"}
2024-04-24 03:15:02 {"level":"info","timestamp":"2024-04-24T01:15:02.991Z","message":"Running on port 8080"}
2024-04-24 03:15:02 {"level":"info","timestamp":"2024-04-24T01:15:02.992Z","message":"Booting"}
2024-04-24 03:15:02 {"level":"info","timestamp":"2024-04-24T01:15:02.996Z","message":"Starting system"}
2024-04-24 03:15:02 {"level":"info","timestamp":"2024-04-24T01:15:02.996Z","message":"System started"}
2024-04-24 03:15:02 {"level":"info","timestamp":"2024-04-24T01:15:02.997Z","message":"Loading sync rules from configuration"}
2024-04-24 03:15:03 {"level":"info","timestamp":"2024-04-24T01:15:03.011Z","message":"Sync rules from configuration unchanged"}
2024-04-24 03:15:03 {"level":"info","timestamp":"2024-04-24T01:15:03.069Z","message":"Created replication slot powersync_2_7ea4 at 00000000/015D1A20 with snapshot 00000003-00000002-1"}
2024-04-24 03:15:03 {"level":"info","timestamp":"2024-04-24T01:15:03.071Z","message":"powersync_2_7ea4 Starting initial replication"}
2024-04-24 03:15:03 {"level":"info","timestamp":"2024-04-24T01:15:03.081Z","message":"powersync_2_7ea4 Replicating \"public\".\"user\""}
2024-04-24 03:15:03 {"level":"error","timestamp":"2024-04-24T01:15:03.084Z","error":"PgError.conn_ended: Unable to do postgres query on ended connection\n    at PgConnection.stream (file:///app/node_modules/.pnpm/github.com+kagis+pgwire@f1cb95f9a0f42a612bb5a6b67bb2eb793fc5fc87/node_modules/pgwire/mod.js:315:13)\n    at stream.next (<anonymous>)\n    at PgResult.fromStream (file:///app/node_modules/.pnpm/github.com+kagis+pgwire@f1cb95f9a0f42a612bb5a6b67bb2eb793fc5fc87/node_modules/pgwire/mod.js:1174:22)\n    at PgConnection.query (file:///app/node_modules/.pnpm/github.com+kagis+pgwire@f1cb95f9a0f42a612bb5a6b67bb2eb793fc5fc87/node_modules/pgwire/mod.js:311:21)\n    at WalStream.startInitialReplication (file:///app/powersync-service/lib/replication/WalStream.js:266:22)\n    at async WalStream.initReplication (file:///app/powersync-service/lib/replication/WalStream.js:442:13)\n    at async WalStream.replicate (file:///app/powersync-service/lib/replication/WalStream.js:431:13)\n    at async WalStreamRunner.replicateOnce (file:///app/powersync-service/lib/replication/WalStreamRunner.js:73:13)\n    at async WalStreamRunner.replicateLoop (file:///app/powersync-service/lib/replication/WalStreamRunner.js:46:13)\n    at async WalStreamRunner.run (file:///app/powersync-service/lib/replication/WalStreamRunner.js:24:13)","message":"Replication error"}
```

```
# See Documentation for more information:
# https://docs.powersync.com/usage/sync-rules
bucket_definitions:
  global:
    data:
      - SELECT * FROM "public"."user"
      - SELECT * FROM "public"."presence"
      - SELECT * FROM "public"."space"
      - SELECT * FROM "public"."room"
      - SELECT * FROM "public"."membership"
      - SELECT * FROM "public"."message"
      - SELECT * FROM "public"."reaction"
```

**Comment by rkistner** [comment]
Discussed with @kobiebotha offline, but noting here:
1. The underlying cause was "-infinity" timestamps breaking replication in PowerSync (not specific to self-hosting).
2. The error triggered a connection failure, which is what was logged, but hid the underlying error. We're improving logging to display the underlying error as well.

**Comment by rkistner** [comment]
The issue is fixed in version 0.17.6:

1. -infinity and infinity are now represented as `0000-01-01 00:00:00Z` and `'9999-12-31 23:59:59Z` respectively (the smallest and largest values supported by sqlite date functions).
2. Error logs should include more details in case of replication failures like these.
